packages/calendar/react/src/calendar.tsx

```diff
@@ -119,26 +88,40 @@ const Calendar: React.FC<Props> = ({
     }
   }, [renderKey, calendarRef]);
 
+  // FIXME: Maybe we should do this on server - check back when payment stuff is done
+  // Verify locale when membership changes
+  const { data: user } = trpc.user.me.useQuery();
+  const { data: preference } = trpc.preference.byCurrentUser.useQuery();
+  const { mutate: updatePreference } = trpc.preference.update.useMutation({
+    onSettled() {
+      utils.user.me.invalidate();
+      utils.preference.byCurrentUser.invalidate();
+    },
+  });
+  useEffect(() => {
+    if (user && user.membership === Membership.FREE) {
+      preference?.formatLanguage !== Language.EN &&
+        updatePreference({
+          formatLanguage: Language.EN,
+        });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [user]);
+
   return (
     <div className="flex h-full">
       <div
-        className="p-8"
+        className="space-y-4 p-8"
         style={{
           width: openPane ? "calc(100% - 325px)" : "100%",
         }}
       >
-        <div className="w-full pb-4">
-          <ControlBar calendarRef={calendarRef} />
-        </div>
-
-        <div className={`w-full ${isFetchingEvents ? "animate-pulse" : ""}`}>
-          <CalendarItem
-            events={events}
-            onRefSettled={setCalendarRef}
-            directoryUsersWithEvents={directoryUsersWithEvents}
-            onPageChange={onPageChange}
-          />
-        </div>
+        <ControlBar calendarRef={calendarRef} />
+
+        <CalendarItem
+          onRefSettled={setCalendarRef}
```

Comment:
ref sets the calendar's ref so that we can reach to the API of the calendar directly

Reply:
> ref sets the calendar's ref so that we can reach to the API of the calendar directly

Is there no type available for this ref? Currently it's any

Reply:
> > ref sets the calendar's ref so that we can reach to the API of the calendar directly
> 
> 
> 
> Is there no type available for this ref? Currently it's any

Yeah there is one for CalendarApi but couldn't find the type for the actual Calendar. Here, we set the calendar reference not api. Api is inside the calendar object.